code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -351,6 +351,81 @@
 	if(!uses)
 		qdel(src)
 
+
+
+/obj/effect/timestop
+	anchored = 1
+	name = "chronofield"
+	desc = "ZA WARUDO"
+	icon = 'icons/effects/160x160.dmi'
+	icon_state = "time"
+	layer = FLY_LAYER
+	pixel_x = -64
+	pixel_y = -64
+	unacidable = 1
+	mouse_opacity = 0
+	var/mob/living/immune = list() // the one who creates the timestop is immune
+	var/list/stopped_atoms = list()
+	var/freezerange = 2
+	var/duration = 140
+	alpha = 125
+
+/obj/effect/timestop/New()
+	..()
+	for(var/mob/living/M in player_list)
+		for(var/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/T in M.mind.spell_list) //People who can stop time are immune to timestop
+			immune |= M
+
+
+/obj/effect/timestop/proc/timestop()
+	playsound(get_turf(src), 'sound/magic/TIMEPARADOX2.ogg', 100, 1, -1)
+	for(var/i in 1 to duration-1)
+		for(var/atom/A in orange (freezerange, src.loc))
```

Comment:
since you don't act on A at all (except further ```istype()```ing), you can skip the typecast yes.